### DISCUSSION TITLE
SmartGit

### DISCUSSION THREAD

**Discussion by ACEngineering** [community/community#28359] [Other Feature Feedback, Questions, & Ideas]
In perusing the internet for a decent Git app, I came across one called SmartGit.  Does anyone have any experience with SmartGit??  I can't seem to be able to get connected to my repositories.  Authentication issues every time, and I've tried several things, but keep coming up empty.

**Comment by Pandapip1** [comment]
I personally recommend VSCode: https://code.visualstudio.com/

**Comment by jfullstackdev** [comment]
GitHub Desktop 

https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/overview/getting-started-with-github-desktop

or GitHub Codespaces

https://docs.github.com/en/codespaces/getting-started/quickstart

Codespaces includes compiling of your code, like Java, 

just a while ago, I finished one of my Java projects without installing Java, Git and Bash locally, it's all through Codespaces, to commit the changes, simply add commit message then click Commit then Sync changes
